### ISSUE TITLE
llvm-nm incorrectly reports length 0 for buffers defined in wasm32 objs

### ISSUE THREAD

**Issue by TerrorJack** [llvm/llvm-project#58839]
Consider this minimal example:

```c
char foo[0x20];
```

If compiled to an ELF object with `cc -c test.c -o test.o` and inspected with `nm -P test.o`, the result will be:

```
foo B 0 20
```

If compiled by clang to a wasm32 object with `clang --target=wasm32-wasi -c test.c -o test.o` and inspected with `llvm-nm -P test.o`, the result will be:

```
foo D 0 0
```

The buffer length is incorrectly reported as `0`. This bug can be reproduced in all recent LLVM major revisions.

**Comment by TerrorJack** [comment]
According to https://github.com/WebAssembly/tool-conventions/blob/main/Linking.md#linking-metadata-section, in a wasm32 object file's symbol table, for any **defined** data symbol, the precise `size` field should be provided. Therefore the information is intact, but `llvm-nm` still doesn't report the correct buffer length.
